pkg/controller/certificatesigningrequests/sync.go

```diff
@@ -124,6 +126,28 @@ func (c *Controller) Sync(ctx context.Context, csr *certificatesv1.CertificateSi
 		}
 	}
 
+	duration, err := pki.DurationFromCertificateSigningRequest(csr)
+	if err != nil {
+		message := fmt.Sprintf("Failed to parse requested duration: %s", err)
+		log.Error(err, message)
+		c.recorder.Event(csr, corev1.EventTypeWarning, "ErrorParseDuration", message)
+		util.CertificateSigningRequestSetFailed(csr, "ErrorParseDuration", message)
+		_, err := util.UpdateOrApplyStatus(ctx, c.certClient, csr, certificatesv1.CertificateFailed, c.fieldManager)
+		return err
+	}
+
+	// Enforce minimum duration of certificate to be 600s to ensure
```

Comment:
While the test asserts 600s, this code technically doesn't, which means when `experimentalapi.CertificateSigningRequestMinimumDuration` changes, this comment is likely to become stale.